src/Entity/Fraction.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mathematicator\Numbers\Entity;
+
+use Brick\Math\BigNumber;
+use Stringable;
+
+/**
+ * Entity to store simple and compound fractions
+ */
+class Fraction
+{
+
+	/** @var Fraction|null */
+	protected $numeratorFraction;
+
+	/** @var string|null */
+	protected $numeratorString;
+
+	/** @var Fraction|null */
+	protected $denominatorFraction;
+
+	/** @var string|null */
+	protected $denominatorString;
+
+	/**
+	 * Superior fraction (if in compound structure)
+	 *
+	 * @var Fraction|null
+	 */
+	protected $parentInNumerator;
+	/**
+	 * Superior fraction (if in compound structure)
+	 *
+	 * @var Fraction|null
+	 */
+	protected $parentInDenominator;
+
+
+	/**
+	 * @param string|Stringable|BigNumber|Fraction|null $numerator optional
+	 * @param string|Stringable|BigNumber|Fraction|null $denominator optional
+	 */
+	public function __construct($numerator = null, $denominator = null)
```

Comment:
Why the input type is not simple `string` only?

Reply:
Because it is not a parser.

Reply:
What is the purpose of this service? Why can't you use tokenization?

Reply:
It is not a service, it's entity representation of a fraction that I want to use for calculator. Simple array is not enough (e.g. for traversing).

Reply:
Tokenizer can be used to create this entity.

Reply:
Ok, thanks. Now I understand.